### PR TITLE
Revert assert

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
@asvetlov added this assert, but we have a test which explicitly says this should work. So, this would be a backwards-incompatible change.

### CODE REVIEW

**Comment by asvetlov** [comment]
Backport error I guess, my wrong.
I'm curious, what is the real type?
If it is `None`, maybe adding `None` type to the union solves the issue.

**Comment by Dreamsorcerer** [comment]
> If it is `None`, maybe adding `None` type to the union solves the issue.

No, I can do an `or x is None` if you'd like to keep the assert.

We also have another problem with the CI though, with the Pypy version being 3.6, which we no longer support.

**Review by asvetlov — APPROVED** [review]
LGTM.
I can live without the assert, not a big deal.

**Comment by Dreamsorcerer** [comment]
Seems like there are more issues with pypy...

**Comment by asvetlov** [comment]
Initialized some `ClientSession` attrs early to allow `__del__` destroy not fully built object by 63217fc.

**Comment by asvetlov** [comment]
Finally, everything is green.
Merging.
@Dreamsorcerer thank you very much for your help!

File: setup.cfg [hunk 1]
```diff
@@ -136,6 +136,7 @@ filterwarnings =
     # Temporarily ignore warnings internal to Python 3.9.7, can be removed again in 3.9.8.
     ignore:The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.:DeprecationWarning:asyncio
     ignore:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning::
+    ignore::ResourceWarning
```
Comment (webknjaz) [inline setup.cfg:139]: This lacks a comment on why this is necessary and under what conditions it can be removed.

**Comment by webknjaz** [comment]
This has been partially ported to 3.8 as https://github.com/aio-libs/aiohttp/commit/78be64bd7e1f3961a4bab75434a997d21a30dcc9.

File: setup.cfg (hunk 1 above)
Comment (webknjaz) [reply setup.cfg:139]: This has now been removed.

**Comment by webknjaz** [comment]
> This has been partially ported to 3.8 as [78be64b](https://github.com/aio-libs/aiohttp/commit/78be64bd7e1f3961a4bab75434a997d21a30dcc9).

And https://github.com/aio-libs/aiohttp/commit/6243204a6a6a0e5ff84ac754218381b44a841e72

**Pull request merged**